missile.py

```diff
@@ -1,23 +1,48 @@
 import pygame
 import math
+from constants import *
 
-class Missile:
-    def __init__(self, x, y, mouse_x, mouse_y):
-        #location
-        self.x = x
-        self.y = y
-        #mouse clicks
-        self.mouse_x = mouse_x
-        self.mouse_y = mouse_y
+class Missile(pygame.sprite.Sprite):
+    def __init__(self, x, y, player):
+        super().__init__()
+
+        #where the bullet starts, turret area of tank image
+        self.x = x + (player.width / 2)
+        self.y = y + (player.height / 2)
+
+        self.image = pygame.Surface((4, 4), pygame.SRCALPHA)
+        self.rect = self.image.get_rect(center=(self.x, self.y))
+        self.color = 'white'
         #speed of missile
         self.speed = 7
-        #math that makes it work
-        self.angle = math.atan2(y-mouse_y, x-mouse_x)
+
+        #sets the direction for where the bullet goes, 
+        #depends fully on the direction the tank is facing
+        if player.direction == TANK_UP:
+            self.direction_x = player.rect.x
+            self.direction_y = player.rect.y - 10
+        elif player.direction == TANK_RIGHT:
+            self.direction_x = player.rect.x +10
+            self.direction_y = player.rect.y
+        elif player.direction == TANK_DOWN:
+            self.direction_x = player.rect.x
+            self.direction_y = player.rect.y +10
+        elif player.direction == TANK_LEFT:
+            self.direction_x = player.rect.x -10
+            self.direction_y = player.rect.y
+
+        #math that makes the bullet move
+        self.angle = math.atan2(y-self.direction_y, x-self.direction_x)
         self.x_vel = math.cos(self.angle) * self.speed
         self.y_vel = math.sin(self.angle) * self.speed
         
-    #display on screen
-    def main(self, display):
+    # display on screen
+    def display(self, display, player):
         self.x -= int(self.x_vel)
         self.y -= int(self.y_vel)
-        pygame.draw.circle(display, (255, 255, 255), (self.x, self.y), 2)
\ No newline at end of file
+        pygame.draw.circle(display, (255, 255, 255), (self.x, self.y), 2)
+
+    #update bullet image while in motion
+    def update(self):
+        self.rect.move_ip(-int(self.x_vel), -int(self.y_vel))
+        pygame.draw.circle(self.image, self.color, self.image.get_rect().center, 2)
```

Comment:
This file was already reviewed in a previous PR....

Reply:
Sorry I think there was an issue when I tried to merge the branch